dio/lib/src/adapters/io_adapter.dart

```diff
@@ -227,7 +238,13 @@ class IOHttpClientAdapter implements HttpClientAdapter {
     );
   }
 
-  HttpClient _configHttpClient(Duration? connectionTimeout) {
+  HttpClient _configHttpClient(Duration? connectionTimeout,
+      {bool newClient = false}) {
```

Comment:
I'd prefer `forceNotCachedClient` for better understanding, or someone else can have better namings.